[R2] Update custom domain docs to align with new UI (#22124)
cloudflare/cloudflare-docs: src/content/docs/r2/buckets/public-buckets.mdx

Public buckets can be set up in either one of two ways:

- Expose your bucket as a custom domain under your control.
- Expose your bucket using a Cloudflare-managed `https://r2.dev` subdomain for non-production use cases.

> These options can be used independently or together, enabling custom domains does not require enabling `r2.dev` access.

Reviewer: ```suggestion
These options can be used independently. Enabling custom domains does not require enabling `r2.dev` access.
```

:::caution

Disable public access to your [`r2.dev` subdomain](#disable-public-development-url) when using products like WAF or Cloudflare Access. If you do not disable public access, your bucket will remain publicly available through your `r2.dev` subdomain.

:::

### Minimum TLS Version

To specify the minimum TLS version of a custom hostname of an R2 bucket, you can issue an API call to edit [R2 custom domain settings](/api/resources/r2/subresources/buckets/subresources/domains/subresources/custom/methods/update/).

## Add your domain to Cloudflare

The domain being used must have been added as a [zone](/fundamentals/setup/accounts-and-zones/#zones) in the same account as the R2 bucket.

> - If your domain is already managed by Cloudflare, you can proceed to Connect a bucket to a custom domain.

Reviewer: ```suggestion
- If your domain is already managed by Cloudflare, you can proceed to [Connect a bucket to a custom domain](/r2/buckets/public-buckets/#connect-a-bucket-to-a-custom-domain).
```

> - If your domain is not managed by Cloudflare, you’ll need to set it up using a [partial (CNAME) setup](/dns/zone-setups/partial-setup/) to add it to your account.

Reviewer: ```suggestion
- If your domain is not managed by Cloudflare, you need to set it up using a [partial (CNAME) setup](/dns/zone-setups/partial-setup/) to add it to your account.
```

:::note

> If the zone is on an Enterprise plan, make sure that you [release the zone hold](/fundamentals/setup/account/account-security/zone-holds/#release-zone-holds) before adding the custom domain.
> A zone hold would prevent the custom subdomain from activating.

Reviewer: ```suggestion
If the zone is on an Enterprise plan, make sure that you [release the zone hold](/fundamentals/setup/account/account-security/zone-holds/#release-zone-holds) before adding the custom domain.

A zone hold would prevent the custom subdomain from activating.
```

To disable a domain:

1. In **R2**, select the bucket you want to modify.
2. On the bucket page, Select **Settings**, go to **Custom Domains**.
3. Next to the domain you want to disable, select **...** and **Disable domain**.
4. The badge under **Access to Bucket** will update to **Not allowed**.

## Remove domain

> Removing a custom domain will disconnect it from your bucket and delete its configuration from the dashboard.
> Your bucket will remain publicly accessible through any other enabled access method, but the domain will no longer appear in the connected domains list.

Reviewer: ```suggestion
Removing a custom domain will disconnect it from your bucket and delete its configuration from the dashboard. Your bucket will remain publicly accessible through any other enabled access method, but the domain will no longer appear in the connected domains list.
```

To remove a domain:

1. In **R2**, select the bucket you want to modify.
2. On the bucket page, Select **Settings**, go to **Custom Domains**.
3. Next to the domain you want to disable, select **...** and **Remove domain**.

> 4. Select ‘Remove domain’ in the confirmation window. The CNAME record pointing to the domain will also be removed as part of this step. You can always add the domain again.

Reviewer: ```suggestion
4. Select ‘Remove domain’ in the confirmation window. This step also removes the CNAME record pointing to the domain. You can always add the domain again.
```

> ## Public Development URL

Reviewer: ```suggestion
## Public development URL
```

> Expose the contents of this R2 bucket to the internet through a Cloudflare-managed r2.dev subdomain.
> This endpoint is intended for non-production traffic.

Reviewer: ```suggestion
Expose the contents of this R2 bucket to the internet through a Cloudflare-managed r2.dev subdomain. This endpoint is intended for non-production traffic.
```

:::note

Public access through `r2.dev` subdomains are rate limited and should only be used for development purposes.

To enable access management, Cache and bot management features, you must set up a custom domain when enabling public access to your bucket.

> Avoid creating a CNAME record pointing to the `r2.dev` subdomain. This is an **unsupported access path**, and we cannot guarantee consistent reliability or performance.
> For production use, [add your domain to Cloudflare](#add-your-domain-to-cloudflare) instead.

Reviewer: ```suggestion
Avoid creating a CNAME record pointing to the `r2.dev` subdomain. This is an **unsupported access path**, and we cannot guarantee consistent reliability or performance. For production use, [add your domain to Cloudflare](#add-your-domain-to-cloudflare) instead.
```

:::

### Enable public development url

When you enable public development URL access for your bucket, its contents become available on the internet through a Cloudflare-managed `r2.dev` subdomain.

To enable access through `r2.dev` for your buckets:

1. In **R2**, select the bucket you want to modify.
2. On the bucket page, select **Settings**.
3. Under **Public Development URL**, select **Enable**.
4. In **Allow Public Access?**, confirm your choice by typing ‘allow’ to confirm and select **Allow**.
5. You can now access the bucket and its objects using the Public Bucket URL.

To verify that your bucket is publicly accessible, check that **Public URL Access** shows **Allowed** in you bucket settings.

### Disable public development url

> Disabling public development URL access removes your bucket’s exposure through the `r2.dev` subdomain.
> The bucket and its objects will no longer be accessible via the Public Bucket URL.
> If you’ve connected other domains, the bucket will remain accessible through those.

Reviewer: ```suggestion
Disabling public development URL access removes your bucket’s exposure through the `r2.dev` subdomain. The bucket and its objects will no longer be accessible via the Public Bucket URL.

If you have connected other domains, the bucket will remain accessible for those domains.
```
